### PR TITLE
[serve] Centralize HTTP-related logic in HTTPState

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Continuation of (and dependent on) https://github.com/ray-project/ray/pull/13018.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@ijrsvt you can wait until the other is merged to review this

**Comment by edoakes** [comment]
@ijrsvt rebased now, PTAL

File: python/ray/serve/controller.py
Comment (ijrsvt) [inline python/ray/serve/controller.py:101]: Could we pull these into a new class like `state.py`?

**Review by ijrsvt — APPROVED** [review]
Looks good! Thanks for doing this!

File: python/ray/serve/controller.py
Comment (edoakes) [reply python/ray/serve/controller.py:101]: Assume you meant new file. Good idea, I'll do that in a follow-up for this and the EndpointState

Comment (ijrsvt) [reply python/ray/serve/controller.py:101]: Oops, yep! Sounds good!

**Pull request merged**
